app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx

```diff
@@ -32,46 +32,51 @@ interface Props {
 }
 
 const ExpirationDatePicker: FunctionComponent<Props> = ({ id, label, tzOffset }) => {
-  const [selectedItem, setSelectedItem] = useState(collection[0])
-  const [pickedDate, setPickedDate] = useState(new Date())
+  const [dropdownSelectedItem, setDropdownSelectedItem] = useState(collection[0])
+  const [calendarPickedDate, setCalendarPickedDate] = useState(new Date())
   const fieldName = `human_${id}`
   const fieldLabel = label ?? 'Expires in'
 
-  const fieldDate = useMemo(() => {
-    if (selectedItem.period === 0) return null
+  const dropdownDate = useMemo(() => {
+    if (dropdownSelectedItem.period === 0) return null
 
-    return new Date(new Date().getTime() + selectedItem.period * dayMs)
-  }, [selectedItem])
+    return new Date(new Date().getTime() + dropdownSelectedItem.period * dayMs)
+  }, [dropdownSelectedItem])
 
-  const dateValue = useMemo(() => {
-    let value = ''
+  const selectedDate= useMemo(() => {
+    let value = null
 
-    if (fieldDate) {
-      value = fieldDate.toISOString()
-    } else if (selectedItem.id === 'custom' ) {
-      value = pickedDate.toISOString()
+    if (dropdownDate) {
+      value = dropdownDate
+    } else if (dropdownSelectedItem.id === 'custom' ) {
+      value = calendarPickedDate
     }
 
     return value
-  }, [fieldDate, selectedItem, pickedDate])
+  }, [dropdownDate, dropdownSelectedItem, calendarPickedDate])
 
   const formattedDateValue = useMemo(() => {
-    if (!dateValue) return
+    if (!selectedDate) return
 
     const formatter = Intl.DateTimeFormat('en-US', {
       month: 'long', day: 'numeric', year: 'numeric', hour: 'numeric', minute: 'numeric', hour12: false
     })
-    const date = new Date(dateValue)
 
-    return formatter.format(date)
-  }, [dateValue])
+    return formatter.format(selectedDate)
+  }, [selectedDate])
 
   const fieldHint = useMemo(() => {
     if (!formattedDateValue) return
 
     return `The token will expire on ${formattedDateValue}`
   }, [formattedDateValue])
 
+  const inputDateValue = useMemo(() => {
```

Comment:
Named like this so it's clear that this is for the input value.

Also, it seems that now variables ending with `Date` are actually Dates (`selectedDate`, `calendarPickedDate`, `dropdownDate`), and the ones ending with `DateValue` are string representaions (`inputDatevalue`, `formattedDateValue`).
This was not on purpose, but I think it makes sense :grimacing: